### ISSUE TITLE
Grabbing Headers Issue

### ISSUE THREAD

**Issue by blimeybloke** [nZEDb/nZEDb#2307]
When grabbing headers from a.b.boneless, the system is not importing anything that is not a .rar format.  I am attaching an nzb file for an example but the files are .10 .11 .12 etc, all have the same subject though its just the extensions are not .rar files.

example filenames:
"a2147176d11ae630e8a424a6f0b981ca.10" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.11" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.12" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.13" yEnc (1/2)
"a2147176d11ae630e8a424a6f0b981ca.14" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.15" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.16" yEnc (1/18)
"a2147176d11ae630e8a424a6f0b981ca.17" yEnc (1/18)

[nzb.txt](https://github.com/nZEDb/nZEDb/files/573152/nzb.txt)

**Comment by blimeybloke** [comment]
Ok, starting to figure this out.  Its adding the post to collections table but with a 0 "totalfiles"

` // Set up the info for inserting into parts/binaries/collections tables.
                        if (!isset($articles[$matches[1]])) {

```
                            // check whether file count should be ignored (XXX packs for now only).
                            $whitelistMatch = false;
                            if ($this->_ignoreFileCount($groupMySQL['name'], $matches[1])) {
                                    $whitelistMatch = true;
                                    $fileCount[1] = $fileCount[3] = 0;
                            }

                            // Attempt to find the file count. If it is not found, set it to 0.
                            if (!$whitelistMatch && !preg_match('/[[(\s](\d{1,5})(\/|[\s_]of[\s_]|-)(\d{1,5})[])\s$:]/i', $matches[1], $fileCount)) {
                                    $fileCount[1] = $fileCount[3] = 0;

                                    if ($this->_showDroppedYEncParts === true) {
                                            file_put_contents(
                                                    nZEDb_LOGS . 'no_files' . $groupMySQL['name'] . '.log',
                                                    $header['Subject'] . PHP_EOL, FILE_APPEND
                                            );
                                    }
                            }
```

`

**Comment by niel** [comment]
Much like windows, nZEDb uses file extensions to identify file types for inclusion. It's very difficult to guess whether something should be included if it uses something non-standard (RAR format stopped using numbers by default many years ago).

**Comment by blimeybloke** [comment]
Am I in the right file though to be trying to edit?  My goal here is to do something that IF there are 0 files, look at the subject to get filecount by comparing for similar subjects.

**Comment by niel** [comment]
I don't know, I don't see a file name mentioned in your post ;-)

**Comment by blimeybloke** [comment]
Sorry about that one.  Something like:
"e2a5ae52de64b19aa0d06491a385aca7.10"
"e2a5ae52de64b19aa0d06491a385aca7.11"
"e2a5ae52de64b19aa0d06491a385aca7.12"

**Comment by niel** [comment]
I'm confused as to what you are asking now. I thought you meant code file to edit, which you didn't supply with your code sample.

**Comment by blimeybloke** [comment]
No I am trying to figure out why the system is not importing files like the ones I listed.  They are mainly in boneless.

**Comment by kevinlekiller** [comment]
There's no file count in the subject so it's not possible to determine how many files there are until a later stage.

**Comment by blimeybloke** [comment]
right, I got that part :) I monitored the parts table, and saw them being added, but then when update_releases runs, it deletes them.  So not sure where to really look to add/modify something for this style of files.

**Comment by kevinlekiller** [comment]
Alright, the issue is somewhere later on, the code you posted doesn't matter for this.

Do you know what is deleting it? This would pin point it more.

**Comment by blimeybloke** [comment]
I followed it this way:  reset the boneless group, imported 1 day (yes it took awhile) then searched for a specific messageID that was part of these files.  Once found, I looked at the binaries table using the binaryID.  Then used the collectionID in the collections table.  All in there but, the collection has totalfiles=0.

This leads me to think that update_releases deletes due to having 0 files.  It is not grouping them.

Hope this helps.

**Comment by kevinlekiller** [comment]
No, the totalfiles is not required, many collections miss the totalfiles.

Usually you see something like this : Some.collection [01/10] "some.file.r01" (01/100)

The [01/10] means that is file 1 of 10, so there's let's say 8 rars and 2 pars, and 01 is r01 and so on.

But the [xx/xx] is not always present, it's not a standard.

**Comment by kevinlekiller** [comment]
The issue is maybe it's not getting properly grouped.

Try this collection regex:

`#^"(?P<match0>[a-fA-F0-9]{32}\.)\d+" yEnc$#`

**Comment by blimeybloke** [comment]
stay tuned.. ill try now

**Comment by kevinlekiller** [comment]
You might have to refresh, I edited it.

**Comment by kevinlekiller** [comment]
This is the edited one in case you missed: `#^"(?P<match0>[a-fA-F0-9]{32}\.)\d+" yEnc$#`

**Comment by blimeybloke** [comment]
I got it. 
![boneless](https://cloud.githubusercontent.com/assets/20652035/20195139/3eb3663c-a763-11e6-8108-992239daa333.png)

Is this correct format?

**Comment by kevinlekiller** [comment]
No, there was an extra `"` at the end I added by mistake, you also need to leave the `#`

**Comment by blimeybloke** [comment]
so literally:
`#^"(?P<match0>[a-fA-F0-9]{32}\.)\d+" yEnc$#`

a # at the beginning and at the end?

**Comment by kevinlekiller** [comment]
Yup. (with the `<match0>`, but github stripped it out, you need to surround it with ` to avoid that on github comments.)

**Comment by blimeybloke** [comment]
Its updating the group but going a bit slower than normal so maybe this is it.. still about 2million articles to go.

**Comment by kevinlekiller** [comment]
Alright.

It won't work on already downloaded articles, you would need to reset the group to do that.

**Comment by blimeybloke** [comment]
yep, I reset and actually purged the group. 

Just ran, update_releases.php 2 true alt.binaries.boneless

same thing, didn't import them:
Process Releases -> Attempting to find complete collections.
199 collections were found to be complete. Time: 0 second(s)

Process Releases -> Calculating collection sizes (in bytes).
199 collections set to filecheck = 3(size calculated)
0 second(s)

Process Releases -> Delete collections smaller/larger than minimum size/file count from group/site setting.
Deleted 223685 collections:
320 smaller than, 221569 bigger than, 1796 with less files than site/group settings in: 1 second(s)

Process Releases -> Create releases from complete collections.
106 Collections ready to be converted to releases.
Added 100 releases.
100 Releases added and 6 duplicate collections deleted in 1 second(s)

**Comment by blimeybloke** [comment]
![boneless](https://cloud.githubusercontent.com/assets/20652035/20195604/3c9c1c02-a765-11e6-8de0-0cf14789d020.png)

that is what I have as the regex btw

**Comment by kevinlekiller** [comment]
Yeah you need to wait a while, since there's no way to know if it's complete, there's no file count, so if it creates a NZB it might be incomplete.

The time is a setting in the admin page.

**Comment by blimeybloke** [comment]
I have the site set to grab the last 1 day of a new group so its grabbing 1 day of boneless.  Or is it another setting?  thank you btw

**Comment by kevinlekiller** [comment]
The time setting is called "delaytime" (in the mysql table), not sure what it's called on the web interface, the default is 2 hours.

For testing, you may want to change the size settings for boneless or globally, the max size settings might be too low, causing it to delete those.

**Comment by kevinlekiller** [comment]
The idea behind the setting is, if the setting were not there, then if there's no file count in the subject, it would create many small NZB's, since we don't know if "a2147176d11ae630e8a424a6f0b981ca.10" is only 1 file in  a set or not, but if we wait, then we get "e2a5ae52de64b19aa0d06491a385aca7.11", and 12 and so on, there could be delays in the person uploading, or in your scripts downloading, or your nntp server propagating, so it waits.

But if it says e2a5ae52de64b19aa0d06491a385aca7.11 [02/40] "e2a5ae52de64b19aa0d06491a385aca7.11", then we know there's 40 files, and if we only have 24 files, then we wait, but as soon as we have 40 we know we have them all so we make the NZB.

**Comment by blimeybloke** [comment]
![min_size](https://cloud.githubusercontent.com/assets/20652035/20195948/83560c88-a766-11e6-95cf-6c0bdb539544.png)

Ill let it go to see what happens.

**Comment by kevinlekiller** [comment]
You could force it to make the NZB if you change the "downloaded" time to a older time in the DB, if you want to know if it's really deleted or not.

**Comment by blimeybloke** [comment]
in the collections table? or binaries?

**Comment by kevinlekiller** [comment]
collections.dateadded to be specific.

You would need to stop update_binaries.php first.

**Comment by blimeybloke** [comment]
thanks, I will give that a shot now

**Comment by blimeybloke** [comment]
nope, the messageid is there, its in the binaries table, but deleted from collections table as soon as the first update_releases.php

**Comment by kevinlekiller** [comment]
This is the column I'm referring to: https://github.com/nZEDb/nZEDb/blob/0.x/resources/db/schema/mysql-ddl.sql#L230

Can you reset the group / do a update_binaries.php then do a select of the collections table and post a log on gist.github.com ?

**Comment by blimeybloke** [comment]
Sure can.  You just need to see the items I am talking about?  Like an export from the table?

**Comment by kevinlekiller** [comment]
Yeah, the ones we are discussing here ideally, maybe do a specific search, like `mysql -u kevin -p -e 'select * from collections where subject like "%a2147176d11ae630e8a424a6f0b981ca%";' nzedb > sql.log`

**Comment by blimeybloke** [comment]
![01parts](https://cloud.githubusercontent.com/assets/20652035/20196567/3a978aaa-a769-11e6-935c-776de96ee31f.png)
![02binaries](https://cloud.githubusercontent.com/assets/20652035/20196566/3a977a38-a769-11e6-9627-9373c6d15589.png)

I am sorry, I just ran it again, there is nothing in the collections table added.

**Comment by blimeybloke** [comment]
Maybe I am looking at this completely wrong.. I have the messageID, but not the subject used in the example nzb file I have.  
[example_nzb.txt](https://github.com/nZEDb/nZEDb/files/584571/example_nzb.txt)

This is quite odd.
If I manually import that, it works and I can see the files.  There is nothing added to the collections table matching the collectionID I have.

**Comment by kevinlekiller** [comment]
I'm not sure what you mean.

**Comment by kevinlekiller** [comment]
The screenshots you posted are of the parts and binaries tables, those are not important, only the collection table is important, it contains most of the metadata.

**Comment by blimeybloke** [comment]
Sorry, how about this image 
![03collection](https://cloud.githubusercontent.com/assets/20652035/20196955/108a14c4-a76b-11e6-8f6e-591e1f99bdab.png)

**Comment by kevinlekiller** [comment]
That looks good, it shows them all grouped.

**Comment by blimeybloke** [comment]
I do see it has total files.  However, its not added when I run update_releases.php, it just gets deleted.

**Comment by kevinlekiller** [comment]
Yeah, the issue is probably those min / max size settings, try putting them all to 0.

**Comment by blimeybloke** [comment]
oh, yes, I see filesize=0 in that field

**Comment by kevinlekiller** [comment]
That's fine for now, it calculates the size later on, the field defaults to 0

**Comment by blimeybloke** [comment]
ok, so in site settings, put Minimum File Size to Make a Release to 0

**Comment by kevinlekiller** [comment]
To save resources it only calculates the size once all files are available.

https://github.com/nZEDb/nZEDb/blob/0.x/nzedb/processing/ProcessReleases.php#L364

**Comment by kevinlekiller** [comment]
You'll have to set that for the group also, and the max size / min files settings also, to make sure.

**Comment by blimeybloke** [comment]
ok, rerunning, will update once a few hours rolls by.

Currently:
38,679 collections waiting to be created (still incomplete or in queue for creation)

I think that is due to the min size=0 but, it did not delete it from the collections table

**Comment by blimeybloke** [comment]
interesting note, the name in collections is:
a8c1d0ea1944adfb98dfeff51af8c2795219587e

but the filenames are:
e2a5ae52de64b19aa0d06491a385aca7

**Comment by kevinlekiller** [comment]
Yeah, that's normal, the collection has a hash to tie all the files together.

**Comment by blimeybloke** [comment]
I changed the date in collections to be: 2016-11-05 03:19:39
It still doesn't seem to want to make it an actual release.

**Comment by kevinlekiller** [comment]
Did you change date or dateadded? I think it's dateadded.

**Comment by blimeybloke** [comment]
no, I did date lol  I just did dateadded and it was removed from collections table but never made it to releases.

**Comment by kevinlekiller** [comment]
Did it say it deleted?

**Comment by blimeybloke** [comment]
I didn't see anything but its not in the either of the tables.

**Comment by kevinlekiller** [comment]
Weird, I'm out of ideas then.

**Comment by blimeybloke** [comment]
I am doing this again, just doing 1 day at a time.
I ran update_binaries on just boneless
then verified it was in collections
ran update_releases on boneless
verified again, its still there
i then updated the dateadded -1 day
ran update_releases on boneless
I saw:
`Process Releases -> Attempting to find complete collections.
1 collections were found to be complete. Time: 0 second(s)

Process Releases -> Calculating collection sizes (in bytes).
1 collections set to filecheck = 3(size calculated)
0 second(s)

Process Releases -> Delete old releases and passworded releases.
Removed releases: 0 past retention, 0 passworded, 0 crossposted, 0 from disabled categories, 0 smaller than category settings, 0 from disabled music genres, 0 from misc->other0 from misc->hashed.
Completed adding 0 releases in 9.90 second(s). 37,190 collections waiting to be created (still incomplete or in queue for creation)
`
verified its still in collections table
reran update_releases and still in collections, but filecheck does equal 3

Should I run something else or do another -1 day to dateadded?

**Comment by blimeybloke** [comment]
i ran update_binaries, update_releases and it was out of collections, but just deleted.

**Comment by blimeybloke** [comment]
Oh, I see why its being deleted but makes no sense:
Process Releases -> Delete collections smaller/larger than minimum size/file count from group/site setting.
Deleted 1 collections:
0 smaller than, 0 bigger than, 1 with less files than site/group settings in: 0 second(s)

the  odd thing is that totalfiles were 21, I did put "50" in filesize.  So it seems as though its being removed due to number of files.

**Comment by blimeybloke** [comment]
The more I dig into this the stranger it is.
So in the parts, it has binaryID of 210092115
in binaries->binaryID=210092115  the collectionID is 94539017

However, collectionID 94539017 does not exist.

When I search in collections based on the name:
SELECT \* FROM `collections` WHERE `subject` LIKE '%a8c1d0ea1944adfb98dfeff51af8c2795219587e%'
I have a collectionID of 94755472

but if I search in binaries for collectionID=94755472 there isnt a match.
